docs/tutorials/build-files-archive-with-dal/using-full-slot.mdx

```diff
@@ -0,0 +1,159 @@
+
+       for page_index in 1..num_pages {
+           let _result = host.reveal_dal_page(
+               target_level as i32,
+               slot_index,
+               page_index.try_into().unwrap(),
+               &mut buffer[page_index as usize * (param.page_size as usize)
+                   ..(page_index as usize + 1) * (param.page_size as usize)],
+           );
+       }
+
+       let hash = blake2b::digest(&buffer, 32).unwrap();
+       let key = hex::encode(hash);
+       let path = OwnedPath::try_from(format!("/{}", key)).unwrap();
+
+       debug_msg!(host, "Saving slot under `{}'\n", path);
+
+       let () = host.store_write_all(&path, &buffer)?;
+
+       Ok(())
+   }
+   ```
+
+   Now the `run` function works like this:
+
+      1. It allocates a buffer of the size of a slot, not a size of a page.
+      1. It tries to fetch the contents of the first page.
+      If 0 bytes are written by `reveal_dal_page`, the targeted slot has not been
+   attested for this block.
+      1. If the targeted slot has been attested, the function reads as many pages as necessary to get the full slot data.
+      1. It stores the data in the durable storage, using the Blake2B hash (encoded in hexadecimal) as its key.
+
+1. Add these `use` statements to the beginning of the file:
+
+   ```rust
+   use tezos_crypto_rs::blake2b;
+   use tezos_smart_rollup::storage::path::OwnedPath;
+   ```
+
+   These dependencies use `tezos_crypto_rs` for hashing, and `hex` for encoding.
+
+1. Add the matching dependencies to the `Cargo.toml` file:
+
+   ```toml
+   tezos_crypto_rs = { version = "0.5.2", default-features = false }
+   hex = "0.4.3"
+   ```
+
+   Adding `default-features = false` for `tezos_crypto_rs` is necessary for the crate to be compatible with Smart Rollups.
+
+1. Deploy the Smart Rollup again, publish a file as you did in the previous section, and wait for enough levels to pass.
+The Smart Rollup log shows the hash of the data, as in this example:
+
+   ```
+   RollupDalParameters { number_of_slots: 32, attestation_lag: 4, slot_size: 65536, page_size: 4096 }
+   Attested slot at index 10 for level 15482
+   Saving slot under `/6a578d1e6746d29243ff81923bcea6375e9344d719ca118e14cd9f3d3b00cd96'
+   See you in the next level
+   ```
+
+1. Get the data from the slot by passing the hash, as in this example:
+
+   ```bash
+   hash=6a578d1e6746d29243ff81923bcea6375e9344d719ca118e14cd9f3d3b00cd96
+   curl "http://localhost:8932/global/block/head/durable/wasm_2_0_0/value?key=/${hash}" \
+       -H 'Content-Type: application/octet-stream' \
+       -o slot.bin
+   ```
+
+1. Inspect the contents of the slot by running this command:
+
+   ```bash
+   hexdump -C slot.bin
+   ```
+
+   For example, if you uploaded a file that contained the text "Hi! This is a message to go on the DAL." the slot might look like this:
+
+   ```
+   00000000  22 34 38 36 39 32 31 32  30 35 34 36 38 36 39 37  |"486921205468697|
```

Comment:
The output of `hexdump` does not look quite right. I’ll have to retry very quickly, but ideally, you should see the text itself.

Reply:
Changed the command to `xxd -r -p slot.bin`, which prints the text.